### PR TITLE
Converts example/flaskr to have a setup.py

### PR DESCRIPTION
**Pull request by wgwz**
Addresses #1902 

Makes the flaskr app easier to run, ex. workflow:
- pip install --editable .
- export FLASK_APP=flaskr.flaskr
- flask initdb
- flask run

Testing is also easier now:
- python setup.py test

### CODE REVIEW

**Comment by wgwz** [comment]
If someone gets a chance to look at this and give me some feedback, I would be glad to convert minitwit into the same format and add that in another commit.

File: examples/flaskr/MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+recursive-include flaskr/templates *
+recursive-include flaskr/static *
```
Comment (davidism) [inline examples/flaskr/MANIFEST.in]:
~~~
graft flaskr/templates
graft flaskr/static
~~~

File: examples/flaskr/MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+include flaskr/flaskr.db
```
Comment (davidism) [inline examples/flaskr/MANIFEST.in]: should not be included in sdist

File: examples/flaskr/README [hunk 3]
```diff
@@ -30,5 +34,5 @@
 
     ~ Is it tested?
 
-      You betcha.  Run the `test_flaskr.py` file to see
+      You betcha.  Run `python setup.py test` to see
```
Comment (davidism) [inline examples/flaskr/README]: Tests should be in project root, not package root.

File: examples/flaskr/MANIFEST.in (hunk 2 above)
Comment (wgwz) [inline examples/flaskr/MANIFEST.in]: what about the schema?

Comment (davidism) [inline examples/flaskr/MANIFEST.in]: That should be.

File: docs/tutorial/testing.rst [hunk 4]
```diff
@@ -8,3 +8,72 @@ expected, it's probably not a bad idea to add automated tests to simplify
...
+
+One possible and recommended project structure is::
+
+    flaskr/
+        flaskr/
+            __init__.py
+            static/
+            templates/
+        tests/
+            __init__.py
```
Comment (davidism) [inline docs/tutorial/testing.rst]: py.test recommends not using `__init__.py`.  `tests` is a directory, not a package.

**Comment by davidism** [comment]
Submitted against 0.11-maintenance in #1945 .

**Comment by gabga** [comment]
Hi,

I am trying to build an application based on the Flaskr tutorial. One of the differences is that I created a sub-package (conveniently called 'utils') that I am trying to access from the application, but here I ran into problems: the application now reports >>ImportError: No module named 'utils'<<

I have looked in the venv to see what has been packaged, but there I have only found an egg-link to the project. When I tested the setup.py manually, everything seemed to be packaged correctly.

Any ideas?

**Comment by davidism** [comment]
@gabga `mypackage.utils`, use absolute imports. Please use Stack Overflow for questions about your own code. This tracker is for issues with Flask's code.

**Comment by gabga** [comment]
Sorry for that. Wgwz asked for any feedback, thought it might be useful for him while also solving my problem.

Anyway, thanks for the help.

**Pull request closed without merging**